ChangeLog.md

```diff
@@ -25,6 +25,10 @@ See docs/process.md for more on how version tagging works.
   in upstream musl code but we don't expect anything major.  Since this is a
   fairly substantial change (at least internally) we are bumping the major
   version of Emscripten to 3. (#13006)
+- Unoptimzed (`-O0`) builds now include the WebAssembly name section by default.
```

Comment:
```suggestion
- Unoptimized (`-O0`) builds now include the WebAssembly name section by default.
```

Reply:
> I think it could make sense to do this in our test suite, but I'm not convinced we should change the general emcc defaults. That seems somewhat risky to me, as now `-O0` would contain significantly more information, and possibly other noticeable differences that people's test suites could notice.

I'm ok with doing for the test suite only (at least initially). 

However, can you be more specific about exactly what kind of users you think could be broken/effected by this change?   Anyone who is monitoring the size of `-O0` builds probably needs to be ready to see increases, no?   That seems like odd things to measure.

Reply:
My understanding from the discussions in #15470 was that we didn't want to add these flags to our test suite only because we wanted to run with default flags.. so I suggest we change the default for `-O0` instead.   Are you now saying you *are* OK with running the tests with non-default flags?

Reply:
Yes, the size probably is not much of a concern. But imagine someone ships an `-O0` build by mistake to production, and now it contains the names section with info they didn't want. Yes, that's a mistake on their part, but they might not even be aware it's a possible type of situation if they haven't read the changelog carefully.

And yes, I was arguing that testing what flags users build with is crucial, but I thought we agreed to a compromise: We still test the flags users do in what should be their production builds, and for debug builds we might as well diverge from them if it improves our testing (like adding the names section, or maybe we'd want to add even more assertions and checks).